Create simple page

olgamie opened the pull request:
Short description (with a reference to an issue).

Simple tutorial on how to create a semantic page:
1) use layout
2) create ui elements
3) use inputs
4) use grid

#208 

**DoD**

- [x] Major project work has a corresponding task. If there’s no task for what you are doing, create it. Each task needs to be well defined and described.

- [x] Change has been tested (manually or with automated tests), everything runs correctly and works as expected. No existing functionality is broken.

- [x] No new error or warning messages are introduced.

- [x] All interaction with a semantic functions, examples and docs are written from the perspective of the person using or receiving it. They are understandable and helpful to this person.

- [x] If the change affects code or repo sctructure, README, documentation and code comments should be updated. 

- [ ] All code has been peer-reviewed before merging into any main branch.

- [ ] All changes have been merged into the main branch we use for development (develop).

- [x] Continuous integration checks (linter, unit tests) are configured and passed.

- [x] Unit tests added for all new or changed logic.

- [ ] All task requirements satisfied. The reviewer is responsible to verify each aspect of the task.

- [x] Any added or touched code follows our style-guide.

dokato commented on vignettes/intro.Rmd:
````diff
@@ -27,6 +28,238 @@ uirender(uiinput(icon("dog"),
 ```
 
+## Create a Shiny App
+Let's create a Shiny app with `semanticPage()` and just a `h1` title.
````
"an `h1`"

olgamie replied:
fixed

dokato commented on vignettes/intro.Rmd:
````diff
@@ -27,6 +28,238 @@ uirender(uiinput(icon("dog"),
+shinyApp(ui, server)
+```
+
+Now let's add sidebar layout `sidebar_layout()`.
````
could you elaborate more, what is the content of sidebar_layout, that it consists of 2 components: sidebar and main, each can take a number of tags, etc.?

olgamie replied:
done

dokato commented on vignettes/intro.Rmd:
````diff
@@ -27,6 +28,238 @@ uirender(uiinput(icon("dog"),
+shinyApp(ui, server)
+```
+
+Next let's fill in the main panel `main_panel()` with a Semantic UI segment containing 2 cards.
````
I think we should keep calling it `Fomantic UI` at this point

olgamie replied:
done

dokato commented on vignettes/intro.Rmd:
````diff
@@ -27,6 +28,238 @@ uirender(uiinput(icon("dog"),
+shinyApp(ui, server)
+```
+
+Time to fill in sidebar `sidebar_panel()`, so let's add dropdown. We create a dropdown input using `dropdown_input()`.
````
"a dropdown" and the "the dropdown", or simply "We create it"

olgamie replied:
done

dokato commented on vignettes/intro.Rmd:
````diff
@@ -27,6 +28,238 @@ uirender(uiinput(icon("dog"),
+Let's create a Shiny app with `semanticPage()` and just a `h1` title.
 
-To define a new page use:
+```{r echo=TRUE, include=TRUE}
````
It applied everywhere, unfortunatelly in the current setting we can't run this markdown as shiny app as we parse vignettes to html for documentation, thus `eval = FALSE` is necessary.

olgamie replied:
done and added screen shots

dokato commented on vignettes/intro.Rmd:
```diff
@@ -27,6 +28,238 @@ uirender(uiinput(icon("dog"),
+)
+
+server <- function(input, output, session) {
+  output$dropdown <- renderUI({
```
why we actually add dropdown with renderUI? if there is a reason please add it to description above

olgamie replied:
done

dokato commented on vignettes/intro.Rmd:
```diff
@@ -27,6 +28,238 @@ uirender(uiinput(icon("dog"),
+    dropdown_input("mtcars_dropdown", c("mpg", "cyl", "disp", "hp"), value = "mpg")
+  })
+
+  data_plot <- reactive({
```
is this really necessary? it might create some confusion, can we just render plot straight from `mtcars[[input$mtcars_dropdown]]`

dokato replied:
alternatively explain this step in details

olgamie replied:
done

dokato commented on vignettes/intro.Rmd:
```diff
@@ -27,6 +28,238 @@ uirender(uiinput(icon("dog"),
+           )
+         )
+      ),
+      plotOutput("plot"),
```
could you rename it as histogram?

olgamie replied:
fixed

dokato commented on vignettes/intro.Rmd:
````diff
@@ -27,6 +28,238 @@ uirender(uiinput(icon("dog"),
+shinyApp(ui, server)
+```
+
+Let's make dropdown do something and generate a histogram from a selected `mtcars` column and a plot.
````
What about: "Let's make dropdown do something. We will generate a histogram from a selected `mtcars` column (look placeholder `histogram`) and also a scatter plot (see placeholder `plot`)."

olgamie replied:
fixed

olgamie commented:
Thanks @dokato for your comments. I addressed remarks in the code. I created a new issue for the tutorial you had in mind. I decided to go with this format as this is sth that other similar packages offer.

krystian8207 reviewed (approved):
lgtm
